### PR TITLE
feat: use system deps

### LINKED ISSUE #26: Support building against system libraries
docling-parse uses cmake's `ExternalProject_Add` feature to download 3rd dependencies at build time. The build system then vendors the 3rd binaries libraries into the wheel. We prefer to use system-provided libraries, so we can track the provenance of all dependencies. System libraries also allow us to address security issues without rebuilding all components.

Would it be possible to add a mode to use system-provided libraries via cmake `find_package()` or `pkgconfig`?

https://github.com/DS4SD/docling-parse/blob/8b7d0d6f9cb19f0c8a7f2871f0a087a90a2158e4/CMakeLists.txt#L100-L105

- utf8cpp has a cmake package provider
- libjpeg-turbo has cmake package and a pkgconfig provider
- nlohmann_json has cmake package and a pkgconfig provider
- libqpdf has a pkgconfig provider

### PR DESCRIPTION
**Pull request by dolfim-ibm**
Resolve #26

### CODE REVIEW

File: src/pybind/docling_parser.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #ifndef PYBIND_PDF_PARSER_H
 #define PYBIND_PDF_PARSER_H
 
-#include <utf8/utf8.h>
+#include <utf8.h>
```
Comment (dolfim-ibm) [inline src/pybind/docling_parser.h:6]: note: this is needed to aligned the include path of the manual externals with the system packages which have `utf8.h` directly in `/usr/include/utf8.h`.

File: app/parse.cpp [hunk 2]
```diff
@@ -12,7 +12,7 @@ Email: {taa, dol, cau, ttt}@zurich.ibm.com
 Date: 18/08/2017
 **********************************************************/
 
-#include <utf8/utf8.h>
+#include <utf8.h>
```
Comment (dolfim-ibm) [inline app/parse.cpp:15]: note: this is needed to aligned the include path of the manual externals with the system packages which have `utf8.h` directly in `/usr/include/utf8.h`.

**Comment by dolfim-ibm** [comment]
Tested on a RHEL VM

```
$ USE_SYSTEM_DEPS=ON poetry install
Installing the current project: docling-parse (1.3.1)

$ poetry run pytest
============================================================== test session starts ===============================================================
platform linux -- Python 3.9.18, pytest-7.4.4, pluggy-1.5.0
rootdir: /data/docling-dev/docling-parse
collected 4 items                                                                                                                                

tests/test_parse.py ....                                                                                                                   [100%]

=============================================================== 4 passed in 12.24s ===============================================================

$ ldd docling_parse/docling_parse.cpython-39-x86_64-linux-gnu.so 
	linux-vdso.so.1 (0x00007ffc2e3bc000)
	libqpdf.so.28 => /lib64/libqpdf.so.28 (0x00007f4d9ecb3000)
	libjpeg.so.62 => /lib64/libjpeg.so.62 (0x00007f4d9ec32000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f4d9ec18000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f4d9ec04000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f4d9e800000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f4d9eb29000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f4d9eb0c000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f4d9e400000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f4d9ef4c000)
	libgnutls.so.30 => /lib64/libgnutls.so.30 (0x00007f4d9e000000)
	libp11-kit.so.0 => /lib64/libp11-kit.so.0 (0x00007f4d9e669000)
	libidn2.so.0 => /lib64/libidn2.so.0 (0x00007f4d9eaeb000)
	libunistring.so.2 => /lib64/libunistring.so.2 (0x00007f4d9e27b000)
	libtasn1.so.6 => /lib64/libtasn1.so.6 (0x00007f4d9ead1000)
	libnettle.so.8 => /lib64/libnettle.so.8 (0x00007f4d9ea7a000)
	libhogweed.so.6 => /lib64/libhogweed.so.6 (0x00007f4d9df68000)
	libffi.so.8 => /lib64/libffi.so.8 (0x00007f4d9ea6e000)

```

**Review by PeterStaar-IBM — DISMISSED** [review]
LGTM!

**Review by PeterStaar-IBM — APPROVED** [review]
great work!!

**Review by PeterStaar-IBM — APPROVED** [review]

**Pull request merged**